src/System.IO.FileSystem/src/System/IO/UnixFileSystem.Linux.cs

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.IO
+{
+    internal sealed partial class UnixFileSystem
+    {
+        public override bool CaseSensitive { get { return true; } }
```

Comment:
Yup, we had a similar issue and punted it until later. It feels like this might be an issue to nail down now that we're hitting it in multiple places.